packages/core/content-manager/admin/src/pages/EditView/components/DocumentActions.tsx

```diff
@@ -673,12 +670,11 @@ const PublishAction: DocumentActionComponent = ({
         },
         transformData(formValues)
       );
-
       if ('data' in res && collectionType !== SINGLE_TYPES) {
         /**
          * TODO: refactor the router so we can just do `../${res.data.documentId}` instead of this.
          */
-        if (id === 'create') {
+        if (id === 'create' && !currentDocumentMeta.documentId) {
```

Comment:
should we maybe retrieve the id from useDocumentContext instead of from useParams? The less we need complex conditions the better